### PR TITLE
Address indexing

### PR DESCRIPTION
**Pull request by tarrenj**
Adds support for Address Indexing, required by Insight nodes. Support should eventually be included in the main release.

Do not merge until @psyraxaus comments approving the merge.

### CODE REVIEW

**Comment by MatteBru** [comment]
Some light documentation about how to modify zen.conf or use flags to start the address indexing would be awesome. Thanks!

**Comment by tarrenj** [comment]
Minor NIT: https://github.com/psyraxaus/zen-addressindexing/issues/1

**Comment by tarrenj** [comment]
NIT resolved, great work @psyraxaus

**Review by albertog78 — APPROVED** [review]

**Review by pierstab — APPROVED** [review]

**Comment by tarrenj** [comment]
Having issues with `qa/rpc-tests/addressindex.py`  Do not merge.

File: qa/rpc-tests/addressindex.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+        self.nodes[0].generate(105)
+        self.sync_all()
+
+        chain_height = self.nodes[1].getblockcount()
+        assert_equal(chain_height, 105)
+        assert_equal(self.nodes[1].getbalance(), 0)
+        assert_equal(self.nodes[2].getbalance(), 0)
+
+        # Check that balances are correct
+        balance0 = self.nodes[1].getaddressbalance("2N2JD6wb56AfK4tfmM6PwdVmoYk2dCKf4Br")
```
Comment (tarrenj) [inline qa/rpc-tests/addressindex.py:49]: Minor NIT:  Testcase fails at this line:
```smrtz@smrtz-VirtualBox:~/builds/zencash/qa/rpc-tests$ ./addressindex.py 
Initializing test directory /tmp/testAFODPK
Mining blocks...
JSONRPC error: Invalid address
  File "/home/smrtz/builds/zencash/qa/rpc-tests/test_framework/test_framework.py", line 123, in main
    self.run_test()
  File "./addressindex.py", line 49, in run_test
    balance0 = self.nodes[1].getaddressbalance("2N2JD6wb56AfK4tfmM6PwdVmoYk2dCKf4Br")
  File "/home/smrtz/builds/zencash/qa/rpc-tests/test_framework/authproxy.py", line 145, in __call__
    raise JSONRPCException(response['error'])
Stopping nodes
Cleaning up
Failed
```

Should be updated to dynamically query the address of each node.

**Comment by tarrenj** [comment]
Reopening PR as the failing test is not a blocking issue.

**Comment by cronicc** [comment]
Closing, see #91

**Pull request closed without merging**